src/key.js

```diff
@@ -771,6 +771,84 @@ Key.prototype.verifyAllUsers = async function(keys) {
   return results;
 };
 
+/**
+ * generate a new subkey to a Key
+ * @param  {Object} options   (optional) options for creating subkey, e.g. {sign: true, passphrase: '123', algorithm:..., curve:...}
+ *
+ * @returns {Promise<module:key.SubKey>} return the subkey if successful.
+ * @async
+ */
+Key.prototype.generateSubkey = async function(options){
+  const defaultOptions = {};
+  if (this.primaryKey.algorithm.indexOf('rsa') === 0) {
+    defaultOptions.numBits = 2048;
+  }
+  else {
+    defaultOptions.curve = enums.curve.ed25519;
+  }
+
+  options = sanitizeKeyOptions(options, defaultOptions);
+  const secretSubKeyPacket = await generateSecretSubkey(options);
+  const result = await this.addSubkey(secretSubKeyPacket, options);
+  return result;
+};
+
+/**
+ * add a subkey packet to a Key
+ * @param {module:packet.PublicSubkey|module:packet.SecretSubkey} the subkey packet to add
+ * @param  {Object} options   (optional) options for subkey, e.g. {sign: true, passphrase: '123'}
+ *
+ * @returns {Promise<module:key.SubKey>} return the subkey if successful.
+ * @async
+ */
+Key.prototype.addSubkey = async function(subkeyPacket, options){
+  if (!this.isPrivate()) {
+    throw new Error("Nothing to addSubkey in a public key");
```

Comment:
"Cannot add a subkey to a public key" would make more sense here

Reply:
Ok, But I just follow the error format of the `Key.prototype.encrypt/decrypt`.